packages/truffle-core/lib/services/google-analytics.js

```diff
@@ -0,0 +1,157 @@
+  /**
+  * set user-level unique id
+  */
+  setUserId: function() {
+    if(!userConfig.get('uniqueId')) {
+      let userId =  nanoid();
+      userConfig.set({ 'uniqueId': userId });
+    } 
+  },
+  /**
+  * set user-level options for analytics
+  * @param {bool} analyticsBool
+  * @param {Object} userConfig
+  */
+  setAnalytics: function(analyticsBool) {
+    if(analyticsBool === true) {
+      this.setUserId();
+      userConfig.set({ 'enableAnalytics': true, 'analyticsSet': true, 'analyticsMessageDateTime': Date.now() });
+    } else {
+      userConfig.set({ 'enableAnalytics': false, 'analyticsSet': true, 'analyticsMessageDateTime': Date.now() });
+    }
+    return true;   
+  },
+  /**
+  * prompt user to determine values for user-level analytics config options
+  * @param {Object} userConfig
+  */
+  setUserConfigViaPrompt: async function() {
+    if(!userConfig.get('analyticsSet') && process.stdin.isTTY === true) {
+      let answer = await inquirer.prompt(analyticsInquiry)
+      if(answer.analyticsInquiry === analyticsInquiry[0].choices[0]) {
+        this.setAnalytics(true);
+      } else {
+        this.setAnalytics(false);
+      }
+    } else if (userConfig.get('analyticsSet') && userConfig.get('enableAnalytics') && process.stdin.isTTY === true) {
+      let answer = await inquirer.prompt(analyticsDisable)
+      if(answer.analyticsDisable) {
+        this.setAnalytics(false); 
+      } else {
+        this.setAnalytics(true);
+      }
+    } else if(userConfig.get('analyticsSet') && !userConfig.get('enableAnalytics') && process.stdin.isTTY === true) {
+      let answer = await inquirer.prompt(analyticsEnable)
+      if(answer.analyticsEnable) {
+        this.setAnalytics(true);
+      } else {
+        this.setAnalytics(false);
+      }
+    } else {
+      return true;
+    }
+    return true;
+  },
+  /**
+  * check user-level config to see if user has enabled analytics
+  * @returns {bool}
+  */
+  checkIfAnalyticsEnabled: function() {
+    if(userConfig.get("enableAnalytics")) {
+      return true;
+    } else {
+      return false;
+    }
+  },
+
+  /**
+  * set data that will be the same in future calls
+  * @returns {Object} visitor
+  */
+  setPersistentAnalyticsData: function() {
+    let visitor = false;
+    if(this.checkIfAnalyticsEnabled() === true) {
+      let visitor = ua(truffleAnalyticsId);
+      let userId = userConfig.get("uniqueId");
+      visitor.set("uid", userId);  
+      return visitor;
+    }
+  },
+
+  /**
+  * send event to Google Analytics
+  * @param {Object}
+  */
+  sendAnalyticsEvent: function(eventObject, callback) {
+    let visitor = this.setPersistentAnalyticsData();
+    if(eventObject["el"]) {
+      eventObject["el"] = eventObject["el"] + " (" + appVersion + ")";
+    } else {
+      eventObject["el"] = appVersion;
+    }
+    if(visitor) {
+      visitor.event(eventObject, function(err) {
+      });
+    }
+    return true;
+  }
+}
+
+process.on('message', function(eventObject) {
```

Comment:
I put this here instead of making a separate executable file. If this is a bad idea and should be put elsewhere please let me know.

Reply:
I really dislike abbreviated variable names :p